Refactor nav to improve responsiveness and tidy up layout bugs

barrymcgee opened the pull request:
## Done

Migrate work carried out on Blues late last year to improve resilience and responsiveness of primary navigation. 

## QA

- Pull code
- Run `./run clean && ./run serve`
- Open http://0.0.0.0:8029
- View the primary nav across breakpoints to ensure responsiveness and resilience.

## Details

Fixes https://github.com/ubuntudesign/juju-squad/issues/575

webteam-app commented:
Starting demo at: https://jaas-ai-canonical-websites-pr-147.run.demo.haus/

hatched commented on templates/shared/header.html:
```diff
@@ -0,0 +1,23 @@
+    <a href="#header-closed" class="p-header__toggle--close" title="Close menu">Close menu</a>
+    <a href="/">
+      <div class="p-header__logo">
+        <svg width="105" height="31"><g transform="translate(.11 .1)" fill="none" fill-rule="evenodd"><circle fill="#E95420" fill-rule="nonzero" cx="15" cy="15" r="15"/><g transform="translate(6.276 2.55)" fill="#FFF" fill-rule="nonzero"><circle cx="9.519" cy="11.538" r="1"/><path d="M10.373 9.83H8.665V3A2.992 2.992 0 0 1 11.654.012 2.992 2.992 0 0 1 14.642 3v1.707h-1.708V3c0-.342-.133-.664-.375- ...
```
Why is this embedded and not a cache-able asset?

hatched commented on templates/shared/search-panel.html:
```diff
@@ -0,0 +1,117 @@
+        <div class="p-search-panel__sidebar">
+          <h4 class="p-search-panel__heading">Popular searches</h4>
+          <ul class="p-list">
+            <li class="p-search-panel__list-item"><a href="{{ url_for('jaasstore.search') }}?q=openstack">openstack</a></li>
```
`url_for` now takes query parameters as an argument.

hatched commented on templates/shared/search-panel.html:
```diff
@@ -0,0 +1,117 @@
+          <ul class="p-inline-list">
+            <li class="p-inline-list__item">
+              <a href="{{ url_for('jaasstore.search') }}?tags=openstack">openstack
+                <span class="p-search-panel__item-count">(108)</span>
```
These counts are weird being hardcoded....are they necessary?

anthonydillon replied:
Yeah, lets remove them as they are never correct.

barrymcgee replied:
Added a separate issue here: https://github.com/canonical-websites/jaas.ai/issues/148

hatched commented on static/sass/custom/_header.scss:
```diff
@@ -0,0 +1,218 @@
+// This breakpoint is here all items in nav will be displayed.
```
I think this needs to be re-worded

anthonydillon commented:
## QA
The navigation does not match the new design very closely: https://zpl.io/anp89LY
- There should be no borders in the navigation
- The selected navigation item is faded
- The navigation items have been changed
- Doesn't seem to use [Vanilla's search pattern](https://docs.vanillaframework.io/en/patterns/search-box#navigation)

barrymcgee commented:
@anthonydillon Addressed your points and added Vanilla search pattern issue as a separate issue here: https://github.com/canonical-websites/jaas.ai/issues/152

anthonydillon commented on templates/shared/_navigation-search.html:
```diff
@@ -0,0 +1,7 @@
+<form data-js="search-form" action="/q" method="GET">
```
This is still not using the Vanilla search pattern.

anthonydillon replied:
Misread you message above, disgard.

clagom commented:
Thanks @barrymcgee, a few comments:
- The active state of the links (e.g. when you open the store or the about page) should be as design: https://zpl.io/anp89LY (Vanilla 2.0). Colour: #9c9c9c
- Mouse over hover colour should be the same as the active colour, seems brighter now? #9c9c9c
- "Getting started" should point to the GUI not to the docs, we need to double check with @hatched @nottrobin 
- The JAAS logo is not aligned to the left, seems 16px too much to the right.
- The "Products" in the global nav is not aligned to the right, seems to be 13px too much to the left. Not sure we need to change that on the Global nav module itself?
- The global nav disappears when in mobile/tablet mode. Is it how it's supposed to be? I'm not sure.
- The "external link" icon seems to big, from Vanilla docs: https://docs.vanillaframework.io/en/patterns/links

clagom commented:
Another thing, @barrymcgee 
In mobile mode, the global nav doesn't work, doesn't open the "projects" panel, see example from jaas.ai and ubuntu.com:

<img width="659" alt="Screenshot 2019-04-10 at 15 12 15" src="https://user-images.githubusercontent.com/1953388/55886307-5e377380-5ba3-11e9-825c-dc180bfbb212.png">

ubuntu.com
<img width="660" alt="Screenshot 2019-04-10 at 15 12 24" src="https://user-images.githubusercontent.com/1953388/55886320-642d5480-5ba3-11e9-8d06-7a7c84ece96a.png">

barrymcgee commented:
@clagom 

- #9c9c9c is not in the [Vanilla color palette](https://docs.vanillaframework.io/en/settings/color-settings)?

- Active and selected state is [a separate PR](https://github.com/canonical-websites/jaas.ai/pull/155) to land after this one

- I've pointed 'Getting started' at the docs as I assumed GUI is in flux and didn't want to point at jujucharms.com - as we still need confirmation, this should not be a show-stopper for this PR but a separate issue when we have the new URL.

hatched commented:
@barrymcgee just FYI there is an 'external link' for the GUI which you can maybe use https://github.com/canonical-websites/jaas.ai/blob/master/webapp/app.py#L59 so you can use the url generator and we only have to update the one spot.

barrymcgee commented:
@clagom 

<img width="1075" alt="Screenshot 2019-04-10 at 16 31 08" src="https://user-images.githubusercontent.com/505570/55892470-46b1b800-5bae-11e9-8497-3a5fdde8be7a.png">

clagom commented:
> @clagom
> 
> <img alt="Screenshot 2019-04-10 at 16 31 08" width="1075" src="https://user-images.githubusercontent.com/505570/55892470-46b1b800-5bae-11e9-8497-3a5fdde8be7a.png">
@barrymcgee 
Perfect, thanks! The rest we discussed in the standup, and the grey that we can use from the current Vanilla is: #666666. Very dark, but the other one available is too light. We'll change it anyway soon with the release of Vanilla 2.0.
